From 8f59727fec1fd37b4c7cd506023cad3c67896523 Mon Sep 17 00:00:00 2001
From: Johan Mossberg <johan.mossberg@intel.com>
Date: Thu, 11 Dec 2014 10:44:44 +0100
Subject: [PATCH 231/236] OV5693: Increase R_RESET (XSHUTDN) to I2C delay to
 30ms

The DS specifies a much shorter delay (1ms +8192 cycles) but even
with the existing 20ms delay we see problems. Sometimes the I2C bus
locks up permanently when starting the camera. Increasing the delay
to 30ms gets rid of this problem.

Change-Id: I93e0784b704483b8a9c629976346b2b564e0a26b
Tracked-On: GMINL-1773
Signed-off-by: Johan Mossberg <johan.mossberg@intel.com>
Reviewed-on: https://icggerrit.ir.intel.com/32384
Reviewed-by: Jing Tao <jing.tao@intel.com>
Approver: Wei Tao <wei.tao@intel.com>
Reviewed-by: Wei Tao <wei.tao@intel.com>
---
 .../external_drivers/camera/drivers/media/i2c/ov5693-ecs/ov5693.c   | 6 ++++--
 1 file changed, 4 insertions(+), 2 deletions(-)

diff --git a/drivers/external_drivers/camera/drivers/media/i2c/ov5693-ecs/ov5693.c b/drivers/external_drivers/camera/drivers/media/i2c/ov5693-ecs/ov5693.c
index 403d6f4..9f5b17a 100644
--- a/drivers/external_drivers/camera/drivers/media/i2c/ov5693-ecs/ov5693.c
+++ b/drivers/external_drivers/camera/drivers/media/i2c/ov5693-ecs/ov5693.c
@@ -1401,8 +1401,10 @@ static int __power_up(struct v4l2_subdev *sd)
 	if (ret)
 		goto fail_clk;
 
-	/* according to DS, 20ms is needed between PWDN and i2c access */
-	msleep(20);
+	/* Value reached through experimentation. The DS specifies a much
+	 * lower value but when using a smaller value the I2C bus sometimes
+	 * locks up permanently when starting the camera. */
+	msleep(30);
 
 	return 0;
 
-- 
1.9.1

